qcodes/instrument_drivers/rigol/DG1062.py

```diff
@@ -0,0 +1,345 @@
+import logging
+from functools import partial
+from typing import Union, Dict
+
+from qcodes import VisaInstrument, validators as vals
+from qcodes import InstrumentChannel, ChannelList
+
+log = logging.getLogger(__name__)
+
+
+def sane_partial(func, docstring, **kwargs):
+    """
+    We want to have a partial function which will allow us access the docstring
+    through the python built-in help function. This is particularly important
+    for client-facing driver methods, whose arguments might not be obvious.
+
+    Consider the follow example why this is needed:
+
+    >>> from functools import partial
+    >>> def f():
+    >>> ... pass
+    >>> g = partial(f)
+    >>> g.__doc__ = "bla"
+    >>> help(g) # this will print an unhelpful message
+
+    Args:
+        func (callable)
+        docstring (str)
+    """
+    ex = partial(func, **kwargs)
+
+    def inner(**inner_kwargs):
+        ex(**inner_kwargs)
+
+    inner.__doc__ = docstring
+
+    return inner
+
+
+class DG1062Burst(InstrumentChannel):
+    """
+    Burst commands for the DG1062. We make a separate channel for these to
+    group burst commands together.
+    """
+
+    def __init__(self, parent: 'DG1062', name: str, channel: int) ->None:
+        super().__init__(parent, name)
+        self.channel = channel
+
+        self.add_parameter(
+            "on",
+            get_cmd=f":SOUR{channel}:BURS?",
+            set_cmd=f":SOUR{channel}:BURS {{}}",
+            vals=vals.Enum(0, 1, "ON", "OFF")
+        )
+
+        self.add_parameter(
+            "polarity",
+            get_cmd=f":SOUR{channel}:BURS:GATE:POL?",
+            set_cmd=f":SOUR{channel}:BURS:GATE:POL {{}}",
```

Comment:
just for my understanding, why are double brackets needed here?

Reply:
that i know, just haven't seen visa commands which contain brackets for the values ;)